docs/Guide.Jest.md

```diff
@@ -3,46 +3,88 @@ id: Guide.Jest
 title: Jest
 ---
 
-## Usage
+## DISCLAIMERS
 
-### 0. Use the [Getting Started](Introduction.GettingStarted.md) Guide to set up Detox
+- This guide describing installing Detox with Jest on a fresh project. If you're migrating an existing project, please apply some common sense in the process.
+- We **strongly** recommend using Jest `24.x.x`.
 
-### 1. Install Jest
+## Installation
+
+### 0. Set up Detox
+
+If you haven't already, use the [Getting Started](Introduction.GettingStarted.md) guide to set up Detox from scratch.
+
+### 1. Install Jest as an npm dependency
 
 ```sh
-npm install --save-dev jest
+> npm install --save-dev jest
 ```
 
-### 2. Run detox init
+### 2. Run Detox init
 
 ```sh
-detox init -r jest
+> detox init -r jest
 ```
 
-Please note, however, that the generated configuration files are not compatible
-with older versions of Jest (23.x or older).
-
-### 3. Modify package.json
-
-```json
-"scripts": {
-    "test:e2e": "detox test -c ios.sim.debug",
-    "test:e2e:build": "detox build"
-},
-"detox": {
-  "test-runner": "jest",
-  "runner-config": "e2e/config.json"
-  ...
-}
+### 3. Fix & Verify
+
+**THIS IS IMPORTANT!** Make sure that `detox init` has ended successfully (everything is green). If it hasn't, be sure to address the issues specified in red. **The ultimate reference is our Jest-based demo app, [`demo-react-native-jest`](https://github.com/wix/Detox/tree/master/examples/demo-react-native-jest)**; Go over these fundamental things:
+
+##### The [*detox* configuration section](https://github.com/wix/Detox/blob/master/examples/demo-react-native-jest/package.json#L25) in your project's `package.json`:
+
+- `test-runner` should be `jest`.
+- `runner-config` to be set to the path of your Jest configuration file (see below). `detox init` generates it as `e2e/config.json`, which is also Jest's default (i.e. used whenever `runner-config` isn't  specified).
+
+![package.json](img/jest-guide/package_json.png)
+
+##### The [Jest configuration file](https://github.com/wix/Detox/blob/master/examples/demo-react-native-jest/e2e/config.json):
+
+- In the very least, verify the following fields: `testEnvironment`, `setupFilesAfterEnv`.
+
+  > Note: `setupFilesAfterEnv` was introduced in Jest 24.
+
+- **`reporters` array field, `verify` bolean field:**
+  We override Jest's default reporters so as to streamline logs of all flavours, in real-time - which isn't Jest's default, thus optimizing it for running e2e tests. This is optional: if you're OK with Jest's default logging, you can either remove `reporters` or set to the `"default"` item. See [Jest's docs](https://jestjs.io/docs/en/configuration#reporters-array-modulename-modulename-options) for more details.
+
+##### The [Jest custom initialization code](https://github.com/wix/Detox/blob/master/examples/demo-react-native-jest/e2e/init.js):
+
+- `beforeAll()`, `beforeEach()`, `afterAll()` should all call our custom Detox-Jest adapter's equivalent lifecycle methods.
```

Comment:
If we'll finally agree on removing `beforeAll()`, then this piece should be updated.